python/dgl/nn/pytorch/explain/pgexplainer.py

```diff
@@ -0,0 +1,363 @@
+            represent the probability of activating the gate for each input element.
+            If training is set to False, the output is also a tensor of probabilities,
+            but they are determined solely by the log_alpha values, without adding any
+            random noise.
+        """
+        if training:
+            bias = self.sample_bias
+            random_noise = torch.rand(w.size()).to(w.device)
+            random_noise = bias + (1 - 2 * bias) * random_noise
+            gate_inputs = torch.log(random_noise) - torch.log(
+                1.0 - random_noise
+            )
+            gate_inputs = (gate_inputs + w) / beta
+            gate_inputs = torch.sigmoid(gate_inputs)
+        else:
+            gate_inputs = torch.sigmoid(w)
+
+        return gate_inputs
+
+    def train_step(self, graph, feat, tmp, **kwargs):
+        r"""Training the explanation network by gradient descent(GD)
+        using Adam optimizer
+
+        Parameters
+        ----------
+        graph : DGLGraph
+            Input batched homogeneous graph.
+        feat : Tensor
+            The input feature of shape :math:`(N, D)`. :math:`N` is the
+            number of nodes, and :math:`D` is the feature size.
+        tmp : float
+            The temperature parameter fed to the sampling procedure.
+        kwargs : dict
+            Additional arguments passed to the GNN model.
+
+        Returns
+        -------
+        Tensor
+            A scalar tensor representing the loss.
+        """
+        self.model = self.model.to(graph.device)
+        self.elayers = self.elayers.to(graph.device)
+
+        pred = self.model(graph, feat, embed=False, **kwargs).argmax(-1).data
+
+        prob, edge_mask = self.explain_graph(
+            graph, feat, tmp=tmp, training=True, **kwargs
+        )
+
+        self.edge_mask = edge_mask
+
+        loss_tmp = self.loss(prob, pred)
+        return loss_tmp
+
+    def explain_graph(self, graph, feat, tmp=1.0, training=False, **kwargs):
+        r"""Learn and return an edge mask that plays a crucial role to
+        explain the prediction made by the GNN for a graph. Also, return
+        the prediction made with the edges chosen based on the edge mask.
+
+        Parameters
+        ----------
+        graph : DGLGraph
+            A homogeneous graph.
+        feat : Tensor
+            The input feature of shape :math:`(N, D)`. :math:`N` is the
+            number of nodes, and :math:`D` is the feature size.
+        tmp : float
+            The temperature parameter fed to the sampling procedure.
+        training : bool
+            Training the explanation network.
+        kwargs : dict
+            Additional arguments passed to the GNN model.
+
+        Returns
+        -------
+        Tensor
+            Classification probabilities given the masked graph. It is a tensor of
+            shape :math:`(1, L)`, where :math:`L` is the different types of label
```

Comment:
@kunmukh I meant these places for "Then you need to cover that in the parameter/returns description. Previously you assumed a single graph there." In particular, this will not be 1, but B for batch size.

Reply:
Understood, fixed.